pennylane/ops/op_math/pow.py

```diff
@@ -345,3 +345,7 @@ def simplify(self) -> Union["Pow", Identity]:
             return op.simplify()
         except PowUndefinedError:
             return Pow(base=base, z=self.z)
+
+    @property
+    def hash(self):
```

Comment:
I don't think we should actually use `self.base.hash` for power.  

Take the case of:
```pycon
>>> op1 = qml.RX(2*np.pi, 0) ** 0.5
>>> op2 = qml.RX(4*np.pi, 0) ** 0.5
>>> qml.math.allclose(op1.matrix(), op2.matrix())
False
>>> op1.hash == op2.hash
True
```

I'm not sure what the best solution to this problem is though.  We encountered a similar problem with `Controlled`.

Maybe turn `hash` into a method and give it an optional keyword?

Reply:
But this is a problem with the rotation gates, right?

```python
>>> op1 = qml.RX(2*np.pi, 0)
>>> op2 = qml.RX(4*np.pi, 0)
>>> qml.math.allclose(op1.matrix(), op2.matrix())
False
>>> op1.hash == op2.hash
True
```

Reply:
Why not take into account the global phase in the `hash` method?

Reply:
For now I just moved the `hash` implementation of the `Controlled` class into the `SymbolicOp` class.

But I am still curious why can't we do this directly in the rotation classes.

Reply:
@AlbertMitjans this is because `hash` is actually with respect to the _expectation_ of a circuit that contains `qml.RX`, so the period will be 2pi rather than 4pi.

It's a holdover from when PL only supported expectations, and is hard to change because this reduces the number of evaluations that the autodiff frameworks request from devices when computing derivatives

Reply:
We seem to have three "modes" of hashing:
1) With respect to expectation values
2) With respect to the state (including global phases, needed by controlled, sprod, sum)
3) With respect to the actual data (needed by pow)

Reply:
Yep. Ideally, `hash` should be with respect to the actual data, and we should have different ways of hashing wrt state and expectations (which the autodiff frameworks will continue to use)

Reply:
> Maybe turn hash into a method and give it an optional keyword?

What do you want to do exactly?

Reply:
When hashing with respect to the actual data, do we want to take into account the interface of the data as well?

Reply:
That's a good question. At the moment we don't, I'm not sure if there are scenarios where that is important or not